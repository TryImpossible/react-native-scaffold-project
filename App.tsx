import RootStack from './src/screens';
export default RootStack;
