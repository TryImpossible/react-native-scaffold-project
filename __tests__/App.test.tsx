import React from 'react';
import renderer from 'react-test-renderer';

import App from '../App';

it('renders correctly with defaults', () => {
  const compoent = renderer.create(<App />);
  let tree = compoent.toJSON();
  expect(tree).toMatchSnapshot();

  tree.children[3].props.onPress(); // 触发点击事件
  tree = compoent.toJSON();
  expect(tree).toMatchSnapshot();
});
