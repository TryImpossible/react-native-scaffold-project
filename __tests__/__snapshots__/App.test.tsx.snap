// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with defaults 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 20,
        "margin": 15,
        "textAlign": "center",
      }
    }
  >
    Welcome to React Native!
  </Text>
  <Text
    style={
      Object {
        "color": "#333333",
        "marginBottom": 5,
        "textAlign": "center",
      }
    }
  >
    To get started, edit App.js
  </Text>
  <Text
    style={
      Object {
        "color": "#333333",
        "marginBottom": 5,
        "textAlign": "center",
      }
    }
  >
    Press Cmd+R to reload,
Cmd+D or shake for dev menu
  </Text>
  <Text
    onPress={[Function]}
    testID="hello_button"
  >
    点我吖
  </Text>
</View>
`;

exports[`renders correctly with defaults 2`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 20,
        "margin": 15,
        "textAlign": "center",
      }
    }
  >
    Welcome to Use Jest!
  </Text>
  <Text
    style={
      Object {
        "color": "#333333",
        "marginBottom": 5,
        "textAlign": "center",
      }
    }
  >
    To get started, edit App.js
  </Text>
  <Text
    style={
      Object {
        "color": "#333333",
        "marginBottom": 5,
        "textAlign": "center",
      }
    }
  >
    Press Cmd+R to reload,
Cmd+D or shake for dev menu
  </Text>
  <Text
    onPress={[Function]}
    testID="hello_button"
  >
    点我吖
  </Text>
</View>
`;
